snuba/query/snql/parser.py

```diff
@@ -81,16 +129,27 @@
     param_expression      = low_pri_arithmetic / quoted_literal
     parameters_list       = parameter* (param_expression)
     parameter             = param_expression space* comma space*
-    function_call         = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)?
+    function_call         = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)? (space* "AS" space* string_literal)?
     simple_term           = quoted_literal / numeric_literal / column_name
     literal               = ~r"[a-zA-Z0-9_\.:-]+"
     quoted_literal        = "'" string_literal "'"
     string_literal        = ~r"[a-zA-Z0-9_\.\+\*\/:-]*"
     numeric_literal       = ~r"-?[0-9]+(\.[0-9]+)?(e[\+\-][0-9]+)?"
+    integer_literal       = ~r"-?[0-9]+"
+    boolean_literal       = true_literal / false_literal
+    true_literal          = ~r"TRUE"i
+    false_literal         = ~r"FALSE"i
     column_name           = ~r"[a-zA-Z_][a-zA-Z0-9_\.]*"
     function_name         = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
+    entity_alias          = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
```

Comment:
if function_name, entity_alias and relationship_name have the same regex, why don't you just declare a symbol regex ? (Entity name can probably be the same)